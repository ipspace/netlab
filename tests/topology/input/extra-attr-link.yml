defaults:
  device: iosv
  extra_attributes:
    link: [ dmz ]

nodes:
  e1:
    igp: [ ospf ]
    edge: true
  e2:
  pe1:
    device: csr
    igp: [ ospf ]

links:
- e1-e2
- e1:
  pe1:
  prefix:
    ipv6: 2001:db8:cafe:1::/64
- e2:
  pe1:
  type: lan
  prefix: 192.168.22.0/24
- e1:
  e2:
  prefix:
    ipv4: 192.168.23.0/24
    ipv6: 2001:db8:cafe:2::/64
  dmz: 100000
