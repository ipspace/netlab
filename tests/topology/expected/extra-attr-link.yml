input:
- topology/input/extra-attr-link.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 10.1.0.1/30
    node: e1
  - ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 10.1.0.2/30
    node: e2
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
- interfaces:
  - ifindex: 2
    ifname: GigabitEthernet0/2
    ipv6: 2001:db8:cafe:1::1/64
    node: e1
  - ifindex: 2
    ifname: GigabitEthernet2
    ipv6: 2001:db8:cafe:1::2/64
    node: pe1
  linkindex: 2
  node_count: 2
  prefix:
    ipv6: 2001:db8:cafe:1::/64
  type: p2p
- bridge: input_3
  interfaces:
  - ifindex: 2
    ifname: GigabitEthernet0/2
    ipv4: 192.168.22.2/24
    node: e2
  - ifindex: 3
    ifname: GigabitEthernet3
    ipv4: 192.168.22.3/24
    node: pe1
  linkindex: 3
  node_count: 2
  prefix:
    ipv4: 192.168.22.0/24
  type: lan
- dmz: 100000
  interfaces:
  - ifindex: 3
    ifname: GigabitEthernet0/3
    ipv4: 192.168.23.1/24
    ipv6: 2001:db8:cafe:2::1/64
    node: e1
  - ifindex: 3
    ifname: GigabitEthernet0/3
    ipv4: 192.168.23.2/24
    ipv6: 2001:db8:cafe:2::2/64
    node: e2
  linkindex: 4
  node_count: 2
  prefix:
    ipv4: 192.168.23.0/24
    ipv6: 2001:db8:cafe:2::/64
  type: p2p
module:
- ospf
name: input
nodes:
  e1:
    af:
      ipv4: true
      ipv6: true
    box: cisco/iosv
    device: iosv
    id: 1
    interfaces:
    - ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 10.1.0.1/30
      linkindex: 1
      name: e1 -> e2
      neighbors:
      - ifname: GigabitEthernet0/1
        ipv4: 10.1.0.2/30
        node: e2
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - ifindex: 2
      ifname: GigabitEthernet0/2
      ipv6: 2001:db8:cafe:1::1/64
      linkindex: 2
      name: e1 -> pe1
      neighbors:
      - ifname: GigabitEthernet2
        ipv6: 2001:db8:cafe:1::2/64
        node: pe1
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - dmz: 100000
      ifindex: 3
      ifname: GigabitEthernet0/3
      ipv4: 192.168.23.1/24
      ipv6: 2001:db8:cafe:2::1/64
      linkindex: 4
      name: e1 -> e2
      neighbors:
      - ifname: GigabitEthernet0/3
        ipv4: 192.168.23.2/24
        ipv6: 2001:db8:cafe:2::2/64
        node: e2
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - ospf
    name: e1
    ospf:
      af:
        ipv4: true
        ipv6: true
      area: 0.0.0.0
      router_id: 10.0.0.1
  e2:
    af:
      ipv4: true
      ipv6: true
    box: cisco/iosv
    device: iosv
    id: 2
    interfaces:
    - ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: e2 -> e1
      neighbors:
      - ifname: GigabitEthernet0/1
        ipv4: 10.1.0.1/30
        node: e1
      type: p2p
    - bridge: input_3
      ifindex: 2
      ifname: GigabitEthernet0/2
      ipv4: 192.168.22.2/24
      linkindex: 3
      name: e2 -> pe1
      neighbors:
      - ifname: GigabitEthernet3
        ipv4: 192.168.22.3/24
        node: pe1
      type: lan
    - dmz: 100000
      ifindex: 3
      ifname: GigabitEthernet0/3
      ipv4: 192.168.23.2/24
      ipv6: 2001:db8:cafe:2::2/64
      linkindex: 4
      name: e2 -> e1
      neighbors:
      - ifname: GigabitEthernet0/3
        ipv4: 192.168.23.1/24
        ipv6: 2001:db8:cafe:2::1/64
        node: e1
      type: p2p
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    name: e2
  pe1:
    af:
      ipv4: true
      ipv6: true
    box: cisco/csr1000v
    device: csr
    id: 3
    interfaces:
    - ifindex: 2
      ifname: GigabitEthernet2
      ipv6: 2001:db8:cafe:1::2/64
      linkindex: 2
      name: pe1 -> e1
      neighbors:
      - ifname: GigabitEthernet0/2
        ipv6: 2001:db8:cafe:1::1/64
        node: e1
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - bridge: input_3
      ifindex: 3
      ifname: GigabitEthernet3
      ipv4: 192.168.22.3/24
      linkindex: 3
      name: pe1 -> e2
      neighbors:
      - ifname: GigabitEthernet0/2
        ipv4: 192.168.22.2/24
        node: e2
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: lan
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: GigabitEthernet1
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    min_mtu: 1500
    module:
    - ospf
    name: pe1
    ospf:
      af:
        ipv4: true
        ipv6: true
      area: 0.0.0.0
      router_id: 10.0.0.3
ospf:
  area: 0.0.0.0
provider: libvirt
